ui/raidboss/data/03-hw/trial/sephirot-ex.ts

```diff
@@ -1,13 +1,322 @@
+import Conditions from '../../../../../resources/conditions';
+import NetRegexes from '../../../../../resources/netregexes';
+import Outputs from '../../../../../resources/outputs';
+import { Responses } from '../../../../../resources/responses';
 import ZoneId from '../../../../../resources/zone_id';
 import { RaidbossData } from '../../../../../types/data';
 import { TriggerSet } from '../../../../../types/trigger';
 
-export type Data = RaidbossData;
+export interface Data extends RaidbossData {
+  mainTank?: string;
+  phase: number;
+  force?: string;
+  shakerTargets?: string[];
+  pillarActive: boolean;
+}
 
 const triggerSet: TriggerSet<Data> = {
   zoneId: ZoneId.ContainmentBayS1T7Extreme,
   timelineFile: 'sephirot-ex.txt',
-  triggers: [],
+  initData: () => {
+    return {
+      phase: 1,
+      pillarActive: false,
+    };
+  },
+  timelineTriggers: [
+    {
+      id: 'SephirotEx Tiferet',
+      regex: /Tiferet/,
+      beforeSeconds: 4,
+      suppressSeconds: 2, // Timeline syncs can otherwise make this extra-noisy
+      response: Responses.aoe(),
+    },
+    {
+      id: 'SephirotEx Triple Trial',
+      regex: /Triple Trial/,
+      beforeSeconds: 4,
+      response: Responses.tankCleave(),
+    },
+    {
+      id: 'SephirotEx Ein Sof Rage',
+      regex: /Ein Sof \(4 puddles\)/,
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Move to safe quadrant',
+        },
+      },
+    },
+    {
+      id: 'SephirotEx Ein Sof Ratzon',
+      regex: /Ein Sof \(1 puddle\)/,
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Bait toward puddle',
+        },
+      },
+    },
+    {
+      id: 'SephirotEx Yesod Bait',
+      regex: /Yesod/,
```

Comment:
Re: pillar presence, the only other thing I can think is to add a counter (and maybe a suppress too, for safety) here?  I'm not sure if that's better or worse, but it's the only alternative I could come up with.

Reply:
Right, that was the other option I considered here, but I *really* don't like using counters unless there's literally no other way. There are 6 Yesods per cycle, and we want to special-case the third one. I *guess* we could start the counter after seeing two Yesods and then check `counter % 6 === 0`, but that only save lines of code, and it wouldn't be super clear to read.